### PR TITLE
Array Intersection

### PR DESCRIPTION
**Pull request by shrutivanw**
CS Fundamentals' assignment to compute the intesection of two input integer arrays (sets).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,12 @@
-# array_intersection
\ No newline at end of file
+# Array Intersection
+Determines interection of two sets of input integer arrays.
```
Comment (Hamled) [inline README.md]: "interection" -> "intersection"

File: README.md [hunk 2]
```diff
@@ -1 +1,12 @@
...
+
+## Exercise
+* Design and implement a method that takes two integer arrays and returns their intersection. The two input arrays have non repeating elements.
+* For example,
+   - intersection of [2, 3, 4] and [4, 5, 6] is [4]
+   - intersection of [50, 43, 25, 72] and [25, 36, 43, 50, 80] is [50, 25, 43]
+   - intersection of [9, 30, 42] and [56, 34, 90, 32] is []
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+**Note**: Do not use Ruby provided functionality.
```
Comment (Hamled) [inline README.md]: We should have an extra blank line above this so that the note goes into its own paragraph.

File: lib/array_intersection.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Creates a new array to return the intersection of the two input arrays
+def intersection(array_1, array_2)
+  puts "NOT IMPLEMENTED"
+  return []
```
Comment (Hamled) [inline lib/array_intersection.rb]: How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

File: specs/array_intersection_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/array_intersection'
+
+describe "array intersection" do
+  describe "basic tests" do
+    describe "test 1" do
```
Comment (Hamled) [inline specs/array_intersection_spec.rb]: Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

File: specs/array_intersection_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+        intersection(array_1, array_2).sort!.must_equal expected_intersection
+      end
+    end
+  end
+
+  # if the parameter is an object, check for nil
+  describe "nil object input test" do
+    describe "nil test 1" do
+      it "first parameter nil" do
```
Comment (Hamled) [inline specs/array_intersection_spec.rb]: I could see this and the other nil test going into the edge cases section.

File: specs/array_intersection_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        array_1 = [70, 90, 34, 21, 78, 42]
+        array_2 = []
+        expected_intersection = []
+
+        intersection(array_1, array_2).sort!.must_equal expected_intersection
+      end
+    end
+
+    describe "edge case 2" do
+      it "first array empty" do
```
Comment (Hamled) [inline specs/array_intersection_spec.rb]: Might be good to also have a test case for when both arrays are empty.

**Review by Hamled — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (shrutivanw) [reply README.md]: Whoops! Now corrected. :)

File: README.md (hunk 2 above)
Comment (shrutivanw) [reply README.md]: done!

File: lib/array_intersection.rb (hunk 3 above)
Comment (shrutivanw) [reply lib/array_intersection.rb]: done!

File: specs/array_intersection_spec.rb (hunk 4 above)
Comment (shrutivanw) [reply specs/array_intersection_spec.rb]: done

File: specs/array_intersection_spec.rb (hunk 5 above)
Comment (shrutivanw) [reply specs/array_intersection_spec.rb]: done

File: specs/array_intersection_spec.rb (hunk 6 above)
Comment (shrutivanw) [reply specs/array_intersection_spec.rb]: done.

**Comment by shrutivanw** [comment]
Feedback incorporated. Thank you @Hamled

**Review by Hamled — APPROVED** [review]
